src/network/networkd-link.c

```diff
@@ -261,6 +261,20 @@ static int link_enable_ipv6(Link *link) {
         return 0;
 }
 
+static bool link_is_bridge_port(Link *link) {
+
+        assert(link);
+
+        if (link->flags & IFF_LOOPBACK)
+                return false;
+
+        if (link->network)
+                if (link->network->bridge)
+                        return true;
```

Comment:
Well basically you want to prevent certain stuff (router discovery and link local address) with link_is_bridge_port() returns true. But the stuff you want to prevent won't happen anyway if it is not network.

For example:

`if (link_ipv6_accept_ra_enabled(link) && !link_is_bridge_port(link))`

`link_ipv6_accept_ra_enabled(link)` already requires `link->network`. So if not "network", it returns false. You don't need link_is_bridge_port to return true.

For the check you added for IPv6LL, is wrapped by

`if (link_ipv6_enabled(link)) {...}`

which:

`return link_dhcp6_enabled(link) || link_ipv6ll_enabled(link) || network_has_static_ipv6_addresses(link->network);`

And all these three requires `link->network` to return true.